### ISSUE TITLE
Add Optimism Ecotone Support

### ISSUE THREAD

**Issue by cgewecke** [cgewecke/hardhat-gas-reporter#214]
PR open for this in #213 but `blobBaseFee`  (necessary to calculate the cost) isn't available from any endpoints yet.

https://github.com/cgewecke/hardhat-gas-reporter/blob/16fde03738b3e37c33f27a98f408b465dc97d4d7/src/utils/gas.ts#L97-L105

A stopgap is to read this off the GasOracle they published yesterday at [0xb528d11cc114e026f138fe568744c6d45ce6da7a][2] on L2

Will have to generate the ABI - it hasn't been verified on the scan. And call with an Alchemy or Infura key.

[Source][1]
```solidity
    /// @notice Retrieves the current blob base fee.
    /// @return Current blob base fee.
    function blobBaseFee() public view returns (uint256) {
        return L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).blobBaseFee();
    }

    /// @notice Retrieves the current base fee scalar.
    /// @return Current base fee scalar.
    function baseFeeScalar() public view returns (uint32) {
        return L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).baseFeeScalar();
    }

    /// @notice Retrieves the current blob base fee scalar.
    /// @return Current blob base fee scalar.
    function blobBaseFeeScalar() public view returns (uint32) {
        return L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).blobBaseFeeScalar();
    }
```

[1]: https://github.com/ethereum-optimism/optimism/blob/e57787ea7d0b9782cea5f32bcb92d0fdeb7bd870/packages/contracts-bedrock/src/L2/GasPriceOracle.sol#L88-L92 
[2]: https://optimistic.etherscan.io/address/0xb528d11cc114e026f138fe568744c6d45ce6da7a

**Comment by cgewecke** [comment]
![Screen Shot 2024-03-14 at 12 48 35 PM](https://github.com/cgewecke/hardhat-gas-reporter/assets/7332026/4948dd7f-b0ac-4cd7-b3d1-d8f458b0c670)

**Comment by cgewecke** [comment]
Closed via release 2.0.1: https://github.com/cgewecke/hardhat-gas-reporter/releases/tag/v2.0.1
